### ISSUE TITLE
TypeError: undefined is not an object (evaluating '_reactNative.NativeModules.AirMapModule.takeSnapshot')

### ISSUE THREAD

**Issue by ccuilla** [react-native-community/react-native-camera#817]
Wondering if anyone has see this issue...

On first try to take a photo I get this:

Android:
```
Possible Unhandled Promise Rejection (id: 0):
TypeError: undefined is not an object (evaluating '_reactNative.NativeModules.AirMapModule.takeSnapshot')
```

iOS:
```
libc++abi.dylib: terminating with uncaught exception of type NSException
```

```
[self.stillImageOutput captureStillImageAsynchronouslyFromConnection:[self.stillImageOutput connectionWithMediaType:AVMediaTypeVideo] completionHandler:^(CMSampleBufferRef imageDataSampleBuffer, NSError *error) {
```

This happens on the first, but then doesn't happen again after reloading the app.

NOTE: This is on react-native 0.46.0. It is also running on a detached Expo v19 app.

**Comment by benjdlambert** [comment]
I think this doesn't look like its anything to do with the `react-native-camera`

I think this is something to do with trying to use `react-native-maps`'s `takeSnapshot()` method on  a ref. `AirMapModule` is from https://github.com/airbnb/react-native-maps.

Unless both modules are clashing?

**Comment by ccuilla** [comment]
Ohhhh. Interesting. We are doing both. So they probably are clashing. Okay. Sorry. Not sure why I assumed this was the camera module.

**Comment by ccuilla** [comment]
Actually, there may be an issue here. We're hanging here in RCTCameraManager.m:

```
      [self.stillImageOutput captureStillImageAsynchronouslyFromConnection:[self.stillImageOutput connectionWithMediaType:AVMediaTypeVideo] completionHandler:^(CMSampleBufferRef imageDataSampleBuffer, NSError *error) {

```

We're getting this even when we have the map snapshot disabled.

**Comment by ccoeder** [comment]
I have iPhone SE with iOS 10.3.3 and im getting this error, is there anyone who can solved it?

**Comment by ccuilla** [comment]
@ccoeder We have not solved it yet. Our team has submitted a PR (I can't seem top find it now). But also forked so we could try to make some progress. But even our fix isn't quite solving the problem completely. It appears to have to do with an instance of RCTCameraManager being created before authorizing the camera access, and the image output variable is initialized at construction time. With this information I tried having the <Camera /> only rendered conditionally once we've obtained permission. That seems to have prevented this crash. But then it only takes one picture. This last issue however may be related to something _else_ in our code. We're still investigating this actively. Pretty critical issue for us.
